src/V3Increments.cpp

```diff
@@ -0,0 +1,222 @@
+        m_insMode = IM_WHILE_PRECOND;
+        m_insStmtp = nodep;
+        iterateAndNextNull(nodep->condp());
+        // Body insert just before themselves
+        m_insStmtp = NULL;  // First thing should be new statement
+        iterateAndNextNull(nodep->bodysp());
+        iterateAndNextNull(nodep->incsp());
+        // Done the loop
+        m_insStmtp = NULL;  // Next thing should be new statement
+    }
+
+    virtual void visit(AstNodeFor* nodep) VL_OVERRIDE {
+        nodep->v3fatalSrc(
+            "For statements should have been converted to while statements in V3Begin.cpp");
+    }
+
+    virtual void visit(AstNodeStmt* nodep) VL_OVERRIDE {
+        if (!nodep->isStatement()) {
+            iterateChildren(nodep);
+            return;
+        }
+        m_insMode = IM_BEFORE;
+        m_insStmtp = nodep;
+        iterateChildren(nodep);
+        m_insStmtp = NULL;  // Next thing should be new statement
+    }
+
+    virtual void visit(AstLogAnd* nodep) VL_OVERRIDE {
+        // PreAdd and Post add not yet supported after &&
+        m_unsupportedHere = true;
+        iterateChildren(nodep);
+        m_unsupportedHere = false;
+    }
+
+    virtual void visit(AstPreAdd* nodep) VL_OVERRIDE {
+        if (m_unsupportedHere) {
+            nodep->v3error("PreAdd operation is not yet supported in this context.\n");
+        }
+        iterateChildren(nodep);
+        AstNodeVarRef* vr = VN_CAST(nodep->op1p(), VarRef);
+        AstConst* constp = VN_CAST(nodep->op2p(), Const);
+        AstConst* newconstp = new AstConst(nodep->fileline(), constp->num());
+        AstNode* backp = nodep->backp();
+
+        // Prepare a temporary variable
+        FileLine* fl = backp->fileline();
+        string name = string("__Vincrement") + cvtToStr(m_modIncrementsNum++);
+        AstVar* varp = new AstVar(fl, AstVarType::BLOCKTEMP, name, vr->varp()->subDTypep());
+
+        // Declare the variable
+        insertBeforeStmt(nodep, varp);
+
+        // Immediately after declaration - increment it by one
+        m_insStmtp->addHereThisAsNext(
+            new AstAssign(fl, new AstVarRef(fl, varp, true),
+                          new AstAdd(fl, new AstVarRef(fl, vr->varp(), false), newconstp)));
+
+        // Immediately after incrementing - assign it to the original variable
+        m_insStmtp->addHereThisAsNext(new AstAssign(fl, new AstVarRef(fl, vr->varp(), true),
+                                                    new AstVarRef(fl, varp, false)));
+
+        // Replace the node with the temporary
+        FileLine* fl_inner = backp->op1p()->fileline();
+        nodep->replaceWith(new AstVarRef(fl_inner, varp, true));
+
+        VL_DO_DANGLING(nodep->deleteTree(), nodep);
+
+        // Both the temporary variable and the original one are now after the
+        // +/- operation. The variable in the original statement is substituted
+        // with the temporary variable.
+    }
+
+    virtual void visit(AstPostAdd* nodep) VL_OVERRIDE {
```

Comment:
```suggestion
    virtual void visit(AstPostAdd* nodep) VL_OVERRIDE {
        visit_prepost(nodep);
    }
```
Then use same code for both pre & post with an if (VN_IS(nodep PostAdd) around only the small code section that differs.